frontend/src/features/nextcloud_integration/api/chat.ts

```diff
@@ -0,0 +1,186 @@
+  const http = createNextcloudAPI();
+  http.defaults.headers['OCS-APIRequest'] = true;
+  http.defaults.headers['Accept'] = 'application/json';
+  http.defaults.headers['format'] = 'json';
+
+  return http.post('/ocs/v2.php/apps/spreed/api/v1/chat/' + chatToken, {
+    message,
+  });
+};
+
+export type TalkConversation = {
+  id: number;
+  token: string;
+  type: number;
+  name: string;
+  displayName: string;
+  objectType: string;
+  objectId: string;
+  participantType: number;
+  participantFlags: number;
+  readOnly: number;
+  hasPassword: boolean;
+  hasCall: boolean;
+  canStartCall: boolean;
+  lastActivity: number;
+  lastReadMessage: number;
+  unreadMessages: number;
+  unreadMention: boolean;
+  unreadMentionDirect: boolean;
+  isFavorite: boolean;
+  canLeaveConversation: boolean;
+  canDeleteConversation: boolean;
+  notificationLevel: number;
+  notificationCalls: number;
+  lobbyState: number;
+  lobbyTimer: number;
+  lastPing: number;
+  sessionId: string;
+  lastMessage: {
+    id: number;
+    token: string;
+    actorType: string;
+    actorId: string;
+    actorDisplayName: string;
+    timestamp: number;
+    message: string;
+    messageParameters: any[];
+    systemMessage: string;
+    messageType: string;
+    isReplyable: boolean;
+    referenceId: string;
+    reactions: Record<string, unknown>;
+    expirationTimestamp: number;
+  };
+  sipEnabled: number;
+  actorType: string;
+  actorId: string;
+  attendeeId: number;
+  permissions: number;
+  attendeePermissions: number;
+  callPermissions: number;
+  defaultPermissions: number;
+  canEnableSIP: boolean;
+  attendeePin: string;
+  description: string;
+  lastCommonReadMessage: number;
+  listable: number;
+  callFlag: number;
+  messageExpiration: number;
+};
+export const getConversations = async (): Promise<Array<TalkConversation>> => {
+  const http = createNextcloudAPI();
+  http.defaults.headers['OCS-APIRequest'] = true;
+
+  const data = await http.get('/ocs/v2.php/apps/spreed/api/v4/room');
+  return data.data.ocs.data;
+};
+
+export enum ConversationType {
+  OneToOne,
+  Group,
+  Public,
+  Changelog,
+  //When a user is deleted from the server or removed from all their conversations, 1 "One to one" rooms are converted to this type)
+  FormerOneToOne,
+}
+export type CreateConversationOptions = {
+  roomType: ConversationType;
+  // User ID (roomType = 1), group ID (roomType = 2 - optional), circle ID (roomType = 2, source = 'circles'], only available with circles-support capability)
+  invite?: string;
+  // The source for the invite, only supported on roomType = 2 for groups and circles (only available with circles-support capability)
+  source?: string;
+  // Conversation name up to 255 characters (Not available for roomType = 1)
+  roomName?: string;
+  // Type of an object this room references, currently only allowed value is 'room' to indicate the parent of a breakout room (See Object types)
```

Comment:
Where is object types?

```suggestion
  // Type of an object this room references, currently only allowed value is 'room' to indicate the parent of a breakout room (see object types)
```

Reply:
Most of the type documentation I took directly from the Nextcloud API docs.
I will add a link to the docs.